### ISSUE TITLE
change pretrained model in coref model

### ISSUE THREAD

**Issue by Sneriko** [allenai/allennlp#4369]
Hi!
I'm trying to build a Swedish coref model using distantly supervised machine learning. The idea is to run AllenNLP:s English coref model on a bilingual dataset (Europarl) and then use a word aligner to transfer the English coreference annotations to swedish annotations and then train the AllenNLP coref model on the Swedish Dataset, using a tiny Swedish coref dataset (17000 tokens, unfortunately the biggest available) as an evaluation set. The transfer is working ok, but just to test the idea I trained the coref model on a very small part of the created Swedish dataset (about 11000 tokens), using a Swedish BERT model as embedding (rather than spanBERT which is used by the allennlp model). I used the same set for evaluation and training (I know this is horrible, but I just wanted to see what happened), and therefore I expected the f1 score to be pretty high, even though the dataset was so small. But I got some strange results. The loss went down, but the f1-score went down as well, all the way to zero, even though the training and evaluation set were the same. Of course, this was just a silly test, and the aim is to create a dataset of about a million tokens, and use the 17000 tokens hand-annotated dataset as an evaluation set, but I was just wondering if there is any logical explanation to these initial results. Is there something I don't understand about the relationship between loss and f1-score?
My second question is whether you actually can just switch the pretrained layer from spanbert to swedish-bert-base and expect the model to be working anyways, although not as accurately?

**Comment by Sneriko** [comment]
Just a clarification, I used the allennlp coref-model both for creating the dataset, and training on this dataset.

**Comment by dirkgr** [comment]
@ZhaofengWu, do you know these answers?

**Comment by ZhaofengWu** [comment]
For your second question, yes you should be able to switch between any huggingface-compatible transformer models and expect the coref model to work. For your first question, in general loss should correlate with F1. I don't know what's happening without delving deeper. I suggest you look at intermediate results of different steps in the coref model and see where the problem is. Is it span filtering, antecedent resolution, or something else? Tokenization is another thing that worths checking.

**Comment by matt-gardner** [comment]
Closing due to lack of activity.
